packages/react-app/src/views/Home.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
+import { Input, Button, notification, Form } from "antd";
+import { useHistory } from "react-router";
+import slugify from "slugify";
+import { ethers } from "ethers";
+
+export default function Admin({ writeContracts, readContracts, admin, mainnetProvider, tx }) {
+  let history = useHistory();
+  const [tipstaCost, setTipstaCost] = useState("...");
+
+  const handleJoinRoom = ({ room }) => {
+    const slugifiedRoom = slugify(room.toLowerCase(), "_");
+
+    history.push(`/room/${slugifiedRoom}`);
+  };
+
+  const becomeDistributor = async () => {
+    const value = ethers.utils.parseEther(tipstaCost);
```

Comment:
Add try catch for situations where user attempts to become a distributor before connecting their wallet

Reply:
I've resolved this... please re-review